### PR TITLE
[cpr] Update cpr to version 1.11.0

### LINKED ISSUE #41876: [cpr] update to 1.11.0
### Library name

cpr

### New version number

1.11.0

### Other information that may be useful (release notes, etc...)

Current latest version in vcpkg is `1.10.5#2` which is according to crp github page is now 'unsupported' version, and it replaced by new `1.11.0` version.
https://github.com/libcpr/cpr/releases/tag/1.11.0

-----

Moreover, `1.11.0` fixes a bug in Android link, which I need.
https://github.com/libcpr/cpr/commit/eb2d418f86d6907d81d67d900e7cd518b2d6e34b#diff-f15f7409d37485ed889fc8730c1f63d6570e55cfb27a4ee9dc7760014a79a739R64

Thanks!

### PR DESCRIPTION
**Pull request by Deishelon**
Fixes: #41876 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Deishelon** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

File: ports/cpr/portfile.cmake [hunk 1]
```diff
@@ -4,10 +4,8 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libcpr/cpr
     REF ${VERSION}
-    SHA512 5e2fe69d5b4dfaa67f636098c8da904b43a22b21cc78bc52446e572ea47f492ce1de0f47fdc2cf34207729ccf007449278f218d8cdeef21f0b98356bca2e5e49
+    SHA512 c314fc576fb8be36bf43326a8a2d8b22d6b2fbb3b494695b84dd8077fc0401981e49890172fc2229d1c68292be2820cd4231d58bcb64326cbe4b73933c092d76
     HEAD_REF master
-    PATCHES
-        disable_werror.patch
```
Comment (dg0yt) [inline ports/cpr/portfile.cmake:10]: Please restore the patch or effect. vcpkg doesn't control the user's toolchain, so it doesn't want `-Werror`.

Comment (Deishelon) [reply ports/cpr/portfile.cmake:10]: Ok, will do, but just FYI, I have not seen port compile fail on Linux (gcc) and Android (Clang). 

Also, I might have missed this in the docs, but general question, why not to turn off these in every possible port? Why need this in this port and not in others ?

Comment (dg0yt) [reply ports/cpr/portfile.cmake:10]: > I have not seen port compile fail on Linux (gcc) and Android (Clang). (CI is also happy)

Sure, but it is only small subset of possible environments. In particular, not including tomorrow's compiler releases' new diagnostic warnings.
And this package even uses `-Wpedantic`, so it is really fragile.

> why not to turn off these in every possible port? Why need this in this port and not in others ?

A patch is needed where a package enables `-Werror` unconditionally. Usually, it is only added when it is noticed, i.e. when it fails for the first user.

Comment (Deishelon) [reply ports/cpr/portfile.cmake:10]: @dg0yt patch is restored (updated to match new CMake file from upstream too)

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x64-windows
x64-windows-static
```

**Pull request merged**